scripts/x-api.js

```diff
@@ -0,0 +1,596 @@
+     * @returns {Object} JSON xAPI statement
+     */
+    self.getArgsXAPIPaused = function (currentTime, duration) {
+      var dateTime = new Date();
+      var timeStamp = dateTime.toISOString();
+
+      var resultExtTime = formatFloat(currentTime);
+
+      var progress = self.getProgress(currentTime, duration);
+
+      playedSegmentsSegmentStart = resultExtTime;
+      endPlayedSegment(resultExtTime);
+
+      var extensions = {};
+      if (typeof resultExtTime !== "undefined") {
+        extensions["https://w3id.org/xapi/video/extensions/time"] = resultExtTime
+      }
+      if (typeof progress !== "undefined") {
+        extensions["https://w3id.org/xapi/video/extensions/progress"] = progress
+      }
+      if (typeof playedSegments !== "undefined") {
+        extensions["https://w3id.org/xapi/video/extensions/played-segments"] = playedSegments
+      }
+
+      return {
+        "verb": {
+          "id": "https://w3id.org/xapi/video/verbs/paused",
+          "display": {
+            "en-US": "paused"
+          }
+        },
+        "object": getXAPIObject(),
+        "result": {
+          "extensions": extensions
+        },
+        "context": {
+          "contextActivities": {
+            "category": [{
+              "id": "https://w3id.org/xapi/video"
+            }]
+          },
+          "extensions": {
+            "https://w3id.org/xapi/video/extensions/session-id": sessionID
+          }
+        },
+        "timestamp" : timeStamp
+      };
+    };
+
+    /**
+     * Generates "seeked" xAPI statement (Video Profile).
+     * @see https://liveaspankaj.gitbooks.io/xapi-video-profile/content/statement_data_model.html#234-seeked
+     *
+     * @public
+     * @param {Number} currentTime time of the video currently
+     * @returns {Object} JSON xAPI statement
+     */
+    self.getArgsXAPISeeked = function (currentTime) {
+      var dateTime = new Date();
+      var timeStamp = dateTime.toISOString();
+      var resultExtTime = formatFloat(currentTime);
+      seekStart = resultExtTime;
+      endPlayedSegment(self.previousTime.toFixed(2));
+      playedSegmentsSegmentStart = seekStart.toFixed(2);
+
+      return {
+        "verb": {
+          "id": "https://w3id.org/xapi/video/verbs/seeked",
+          "display": {
+            "en-US": "seeked"
+          }
+        },
+        "object": getXAPIObject(),
+        "result": {
+          "extensions" : {
+            "https://w3id.org/xapi/video/extensions/time-from": self.previousTime.toFixed(2),
+            "https://w3id.org/xapi/video/extensions/time-to": seekStart.toFixed(2)
+          }
+        },
+        "context": {
+          "contextActivities": {
+            "category": [{
+              "id": "https://w3id.org/xapi/video"
+            }]
+          },
+          "extensions": {
+            "https://w3id.org/xapi/video/extensions/session-id": sessionID
+          }
+        },
+        "timestamp" : timeStamp
+      };
+    };
+
+    /**
```

Comment:
Re: `interacted`, I think I'd prefer to leave it, just because all the different types of interactions have such different `extension` values. I could refactor out the common components into a private function called `getArgsXAPIInteracted()`, but then I'd be splitting up the xAPI functions between public and private, and all the JSON data wouldn't be together, etc.

Maybe reevaluate in the future if additional interactions get implemented?